src/profiling/with_perf.md

```diff
@@ -0,0 +1,294 @@
+# Profiling with perf
+
+sThis is a guide for how to profile rustc with perf.
+
+## Initial steps
+
+- Get a clean checkout of rust-lang/master, or whatever it is you want to profile.
+- Set the following settings in your `config.toml`:
+  - `debuginfo-lines = true`
+  - `use-jemalloc = false` -- lets you do memory use profiling with valgrind 
```

Comment:
nit: we have been trying to move away from `--` and towards em-dash. Also, elsewhere.

Reply:
oh? doesn't markdown change `--` to em-dash? em-dashes are annoying to type :)

Reply:
I guess there is some HTML entity for them...

Reply:
OK, I resorted to copy-and-paste :)

Reply:
At least on GitHub, they are not changed. I find them annoying too, but we've already changed them elsewhere. I usually copy/paste too.

Thanks :)